storage-provider/server/src/p2p.rs

```diff
@@ -0,0 +1,238 @@
+use std::{fmt::Display, path::PathBuf, str::FromStr};
+
+use bootstrap::{BootstrapBehaviour, BootstrapBehaviourEvent};
+use clap::ValueEnum;
+use ed25519_dalek::{pkcs8::DecodePrivateKey, SigningKey};
+use libp2p::{
+    futures::StreamExt, identify, identity::Keypair, rendezvous, rendezvous::Namespace,
+    swarm::SwarmEvent, Multiaddr, PeerId, Swarm,
+};
+use register::{RegisterBehaviour, RegisterBehaviourEvent};
+use serde::de;
+use tokio_util::{sync::CancellationToken, task::TaskTracker};
+use tracing::{error, info};
+
+mod bootstrap;
+mod register;
+
+pub(crate) use bootstrap::BootstrapConfig;
+pub(crate) use register::RegisterConfig;
+
+const P2P_NAMESPACE: &str = "polka-storage";
+
+#[derive(Debug, Clone, Copy, ValueEnum)]
+pub enum NodeType {
+    Bootstrap,
+    Register,
+}
+
+impl Display for NodeType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
```

Comment:
I feel that this makes the code more explicit
```suggestion
        Debug::fmt(&self, f)
```